model/transaction.go

```diff
@@ -56,6 +56,12 @@ type Transaction struct {
 	// duration metrics.
 	DurationHistogram Histogram
 
+	// DurationAggregate stores the value_count, min, max and sum of observed duration for the service metrics.
+	DurationAggregate AggregateMetric
+
+	// FailureCount is used by service metrics documents to indicate how many failed transactions were observed
+	FailureCount *int
```

Comment:
We have `event.outcome`, which we group by in transaction metrics to calculate the failed transaction rate. Can we do the same here?